adapters/repos/db/index.go

```diff
@@ -1804,35 +1800,19 @@ func (i *Index) initLocalShardWithForcedLoading(ctx context.Context, class *mode
 	return nil
 }
 
-func (i *Index) getLocalShardNoShutdown(shardName string) (
+func (i *Index) GetShard(ctx context.Context, shardName string) (
 	shard ShardLike, release func(), err error,
 ) {
-	i.closeLock.RLock()
-	defer i.closeLock.RUnlock()
-
-	if i.closed {
-		return nil, func() {}, errAlreadyShutdown
-	}
-
-	// make sure same shard is not inited in parallel
-	i.shardCreateLocks.Lock(shardName)
-	defer i.shardCreateLocks.Unlock(shardName)
-
-	// check if created in the meantime by concurrent call
-	shard = i.shards.Load(shardName)
-	if shard == nil {
-		return nil, func() {}, nil
-	}
-
-	release, err = shard.preventShutdown()
-	if err != nil {
-		return nil, func() {}, fmt.Errorf("get/init local shard %q, no shutdown: %w", shardName, err)
-	}
+	return i.getOptInitLocalShard(ctx, shardName, false)
+}
 
-	return shard, release, nil
+func (i *Index) getOrInitShard(ctx context.Context, shardName string) (
+	shard ShardLike, release func(), err error,
+) {
+	return i.getOptInitLocalShard(ctx, shardName, true)
 }
 
-func (i *Index) getOrInitLocalShardNoShutdown(ctx context.Context, shardName string) (
+func (i *Index) getOptInitLocalShard(ctx context.Context, shardName string, ensureInit bool) (
```

Comment:
Can we detail here in the godoc exactly what ensureInit means, and also how to handle the `release` function, what does it entails ?